core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -1653,8 +1660,7 @@ class KafkaApis(val requestChannel: RequestChannel,
           (txnCoordinator.partitionFor(key), TRANSACTION_STATE_TOPIC_NAME)
 
         case CoordinatorType.SHARE =>
-          // None check already done above
-          (shareCoordinator.get.partitionFor(key), SHARE_GROUP_STATE_TOPIC_NAME)
+          (shareCoordinator.foreach(coordinator => coordinator.partitionFor(SharePartitionKey.getInstance(key))), SHARE_GROUP_STATE_TOPIC_NAME)
```

Comment:
For my understanding, why did we change from get to foreach?

Reply:
This is the idiomatic way to handle Optionals in scala

https://www.scala-lang.org/api/2.12.7/scala/Option.html

https://stackoverflow.com/questions/6819373/why-is-foreach-better-than-get-for-scala-options